src/privileged/blurts/FirefoxMonitor.jsm

```diff
@@ -0,0 +1,834 @@
+        elt.addEventListener("keydown", function listener(event) {
+
+          handleInputs(event, "2", elt, doc);
+
+          // elt.removeEventListener("keydown", listener);
+          // let popupNotification = doc.getElementById("breach-alerts-notification");
+          // doc.getAnonymousElementByAttribute(popupNotification, "anonid", "button").click();
+        });
+        this._textbox = elt;
+        box.appendChild(elt);
+        return box;
+      },
+    };
+    retval.primaryAction = {
+      label: "Search Firefox Monitor",
+      accessKey: "f",
+      callback: function() {
+        FirefoxMonitor.notifyEventListeners(`variant_2_submit`);
+        let stringStream = Cc["@mozilla.org/io/string-input-stream;1"].
+          createInstance(Ci.nsIStringInputStream);
+        stringStream.data = `emailHash=${sha1(this._textbox.value)}`;
+
+        let postData = Cc["@mozilla.org/network/mime-input-stream;1"].
+          createInstance(Ci.nsIMIMEInputStream);
+        postData.addHeader("Content-Type", "application/x-www-form-urlencoded");
+        postData.setData(stringStream);
+        doc.defaultView.openUILinkIn("https://monitor.firefox.com/scan", "tab", { postData });
+      }.bind(retval),
+    };
+    retval.secondaryActions = [
+      {
+        label: "Dismiss",
+        accessKey: "d",
+        callback: () => {
+          FirefoxMonitor.notifyEventListeners(`variant_2_dismiss`);
+          showSurvey(browser, true);
+        },
+      }, {
+        label: "Never show breach alerts",
+        accessKey: "n",
+        callback: () => {
+          FirefoxMonitor.notifyEventListeners(`variant_2_dismiss_permanent`);
+          blurtsDisabled = true;
+          showSurvey(browser, true);
+        },
+      }
+    ];
+    return retval;
+  },
+
+  //variant #3
+  function(browser, doc, host, site) {
+    let retval = {
+      _textbox: null,
+      _checkbox: null,
+      get box() {
+        let box = doc.createElementNS(XUL_NS, "vbox");
+
+        let elt = doc.createElementNS(XUL_NS, "hbox");
+        elt.setAttribute("style", "margin-bottom: 1rem;");
+        elt.setAttribute("pack", "center");
+        let imgElt = doc.createElementNS(XUL_NS, "image");
+        imgElt.setAttribute("src", imageDataURIs.iconLaptopRisk);
+        imgElt.setAttribute("style", "width: 148px;");
+        elt.appendChild(imgElt);
+        box.appendChild(elt);
+        elt = doc.createElementNS(XUL_NS, "description");
+        elt.setAttribute("anonid", "maindesc");
+        elt.appendChild(doc.createTextNode("Have an account on this website?\nIt may be at risk."));
+        elt.setAttribute("style", "font-size: 150%; text-align: center; white-space: pre;");
+        box.appendChild(elt);
+        elt = doc.createElementNS(XUL_NS, "description");
+        let strings = [
+          {str: site.Title + " was reported to "},
+          {str: "Firefox Monitor", link: `https://monitor.firefox.com/?breach=${host}`},
```

Comment:
I can’t think of cases off the top of my head, but should the host in the query string be urlencoded? Trying to think if a Unicode url could break this or not. We’d need to make the change in several places (which also means it could be a bit D.R.Y.er).